plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Parse.hs

```diff
@@ -0,0 +1,46 @@
+module Ide.Plugin.Cabal.Parse
+( parseCabalFile
+, parseCabalFileContents
+  -- * Re-exports
+, FilePath
+, NonEmpty(..)
+, PWarning(..)
+, Version
+, PError(..)
+, Position(..)
+, GenericPackageDescription(..)
+) where
+
+import           Control.Monad                                (unless)
+import qualified Data.ByteString                              as BS
+import           Data.List.NonEmpty                           (NonEmpty (..))
+import           Distribution.Fields                          (PError (..),
+                                                               PWarning (..))
+import           Distribution.Fields.ParseResult              (runParseResult)
+import           Distribution.PackageDescription.Parsec       (parseGenericPackageDescription)
+import           Distribution.Parsec.Position                 (Position (..))
+import           Distribution.Types.GenericPackageDescription (GenericPackageDescription (..))
+import           Distribution.Types.Version                   (Version)
+import qualified System.Directory                             as Dir
+import qualified System.Exit                                  as Exit
+
+
+parseCabalFile
+    :: FilePath
+    -> IO ([PWarning], Either (Maybe Version, NonEmpty PError) GenericPackageDescription)
+parseCabalFile =
+    readAndParseFile'
+  where
+    readAndParseFile' fpath = do
+        exists <- Dir.doesFileExist fpath
+        unless exists $
+            Exit.die $
```

Comment:
Surely not! Won't this kill HLS??? We definitely need to forbid this function in hlint lol

Reply:
LOL, good catch

Reply:
Code wasn't used :angel: